src/framework/script/script-attributes.js

```diff
@@ -148,6 +146,53 @@ function rawToValue(app, args, value, old) {
     return value;
 }
 
+/**
+ * @typedef {Object} AttributeSchema
+ * @property {"boolean"|"number"|"string"|"json"|"asset"|"entity"|"rgb"|"rgba"|"vec2"|"vec3"|"vec4"|"curve"} type - The Attribute type
+ * @property {boolean} [array] - True if this attribute is an array of `type`
+ */
+
+/**
+ * Takes an attribute schema, a value and current value, and return a new value
+ *
+ * @param {import('../../framework/application.js').Application} app - The working application
+ * @param {AttributeSchema} schema - The attribute schema used to resolve properties
+ * @param {*} value - The raw value to create
+ * @param {*} current - The existing value
+ * @returns {*} The return value
+ */
+export function attributeToValue(app, schema, value, current) {
+    if (schema.array) {
+        const arr = [];
+        for (let i = 0, len = value.length; i < len; i++) {
+            arr.push(rawToValue(app, schema, value[i], current ? current[i] : null));
+        }
+        return arr;
+    }
+
+    return rawToValue(app, schema, value, current);
+}
+
+/**
+ * This will assign values to a script instance based on a map of attributes schemas and a corresponding map of data
```

Comment:
Limit JSDoc blocks to 100 characters in width unless there's a good reason not to. And end sentences with a full stop.